src/main/java/com/checkmarx/flow/utils/ScanUtils.java

```diff
@@ -751,16 +691,167 @@ public static String getHTMLBody(ScanResults.XIssue issue, ScanRequest request,
         String branch = request.getBranch();
         StringBuilder body = new StringBuilder();
         body.append("<div>");
+
+        if (Optional.ofNullable(issue.getScaDetails()).isPresent()) {
+            setSCAHtmlBody(issue, request, body);
+
+        } else {
+            setSASTHtmlBody(issue, flowProperties, branch, body);
+        }
+        body.append(DIV_CLOSING_TAG);
+        return body.toString();
+    }
+
+    /**
+     *
+     * @param request   The scanRequest object
+     * @param issue     The scanResults issue
+     * @param extraTags Extra tags array. Jira issue prefix/postfix are on the [0], [1] positions
+     * @return  Issue key according to the bug type parameter
+     */
+    public static String getScaSummaryIssueKey(ScanRequest request, ScanResults.XIssue issue, String... extraTags) {
+        ScanResults.ScaDetails scaDetails = issue.getScaDetails().get(0);
+        String bugType = request.getBugTracker().getType().getType();
+
+        switch (bugType) {
+            case "JIRA":
+                String issuePrefix = extraTags[0];
+                String issuePostfix = extraTags[1];
+
+                Finding detailsFindings = scaDetails.getFinding();
+                Package vulnerabilityPackage = scaDetails.getVulnerabilityPackage();
+
+                return getJiraScaSummaryIssueKey(request, issuePrefix, issuePostfix, detailsFindings, vulnerabilityPackage);
+            case "CUSTOM":
+                return getCustomScaSummaryIssueKey(request, scaDetails);
+            default:
+                throw new NotImplementedException("Summary issue key wasn't implemented yet for bug type: {}", bugType);
+        }
+    }
+
+    private static void setSASTMDBody(ScanResults.XIssue issue, String branch, String fileUrl, FlowProperties flowProperties, StringBuilder body) {
```

Comment:
What about extracting these formatting methods into classes? ScanUtils is already more than 1000 lines long.
E.g. ScaMessageFormatter/SastMessageFormatter, or MarkdownFormatter/HtmlFormatter.

Reply:
I talked about this with Aviv and he said he opened some relevant tasks including this issue.